homeassistant/components/rejseplanen/sensor.py

```diff
@@ -6,252 +6,267 @@
+    ATTR_STOP_NAME,
+    ATTR_TRACK,
+    CONF_DEPARTURE_TYPE,
+    CONF_DIRECTION,
+    CONF_NAME,
+    CONF_ROUTE,
+    CONF_STOP_ID,
+    DEFAULT_STOP_NAME,
+    DOMAIN,
 )
-from homeassistant.const import CONF_NAME, UnitOfTime
-from homeassistant.core import HomeAssistant
-from homeassistant.helpers import config_validation as cv
-from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
-from homeassistant.util import dt as dt_util
+from .coordinator import RejseplanenDataUpdateCoordinator
+from .entity import RejseplanenUpdaterStatusSensor
 
 _LOGGER = logging.getLogger(__name__)
 
-ATTR_STOP_ID = "stop_id"
-ATTR_STOP_NAME = "stop"
-ATTR_ROUTE = "route"
-ATTR_TYPE = "type"
-ATTR_DIRECTION = "direction"
-ATTR_FINAL_STOP = "final_stop"
-ATTR_DUE_IN = "due_in"
-ATTR_DUE_AT = "due_at"
-ATTR_SCHEDULED_AT = "scheduled_at"
-ATTR_REAL_TIME_AT = "real_time_at"
-ATTR_TRACK = "track"
-ATTR_NEXT_UP = "next_departures"
-
-CONF_STOP_ID = "stop_id"
-CONF_ROUTE = "route"
-CONF_DIRECTION = "direction"
-CONF_DEPARTURE_TYPE = "departure_type"
-
-DEFAULT_NAME = "Next departure"
-
-
-SCAN_INTERVAL = timedelta(minutes=1)
-
-BUS_TYPES = ["BUS", "EXB", "TB"]
-TRAIN_TYPES = ["LET", "S", "REG", "IC", "LYN", "TOG"]
-METRO_TYPES = ["M"]
-
-PLATFORM_SCHEMA = SENSOR_PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_STOP_ID): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_ROUTE, default=[]): vol.All(cv.ensure_list, [cv.string]),
-        vol.Optional(CONF_DIRECTION, default=[]): vol.All(cv.ensure_list, [cv.string]),
-        vol.Optional(CONF_DEPARTURE_TYPE, default=[]): vol.All(
-            cv.ensure_list, [vol.In([*BUS_TYPES, *TRAIN_TYPES, *METRO_TYPES])]
-        ),
-    }
-)
-
-
-def due_in_minutes(timestamp):
-    """Get the time in minutes from a timestamp.
-
-    The timestamp should be in the format day.month.year hour:minute
-    """
-    diff = datetime.strptime(timestamp, "%d.%m.%y %H:%M") - dt_util.now().replace(
-        tzinfo=None
-    )
-
-    return int(diff.total_seconds() // 60)
-
 
-def setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    add_devices: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+    config_entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
 ) -> None:
     """Set up the Rejseplanen transport sensor."""
-    name = config[CONF_NAME]
-    stop_id = config[CONF_STOP_ID]
-    route = config.get(CONF_ROUTE)
-    direction = config[CONF_DIRECTION]
-    departure_type = config[CONF_DEPARTURE_TYPE]
-
-    data = PublicTransportData(stop_id, route, direction, departure_type)
-    add_devices(
-        [RejseplanenTransportSensor(data, stop_id, route, direction, name)], True
+
+    _LOGGER.info(
+        "Setting up Rejseplanen transport sensor for entry: %s", config_entry.entry_id
     )
```

Comment:
So we need to keep this intact and raise an issue explaining users that the YAML is deprecated and how they can set up again